capellambse/diagram_cache.py

```diff
@@ -0,0 +1,83 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""CLI for the diagram cache updating feature."""
+from __future__ import annotations
+
+import pathlib
+import sys
+
+import capellambse
+from capellambse import _diagram_cache, cli_helpers
+
+try:
+    import click
+except ImportError:
+
+    def _main() -> None:
+        """Display a dependency error."""
+        print("Missing dependency 'capellambse[cli]'", file=sys.stderr)
+        raise SystemExit(1)
+
+else:
+
+    @click.command()
+    @click.option(
+        "-m",
+        "--model",
+        "model_",
+        type=cli_helpers.ModelCLI(),
+        required=True,
+        help="Capella model to export diagrams from",
+    )
+    @click.option(
+        "-o",
+        "--output",
+        type=click.Path(
+            file_okay=False, dir_okay=True, path_type=pathlib.Path
+        ),
+        default="diagrams",
+        help="Directory to store the rendered diagrams in",
+        show_defaults=True,
+    )
+    @click.option(
+        "-f",
+        "--format",
+        default="svg",
+        help="Image output format",
+        show_defaults=True,
+    )
```

Comment:
I just saw another Github conversation about allowed formats of the diagram cache. I'd use `click.Choice` to enforce the available options with `click`.

Reply:
The idea behind allowing any string here was to later let Capella decide whether it knows that format or not; this way Capella can add other image formats in the future without us having to explicitly add them to the Choice options. Do you think that makes sense? I'm also fine with switching to Choice for faster (and probably more useful) error messages.

Reply:
I'd hard code the choices following Moritz' proposal. I don't expect Obeo to do something here. And if they really add/ remove an image format it is a very tiny and quick change we can apply on our side.

Reply:
@Wuestengecko I can understand the point, but the usability improvement (fail fast principle and better error message, the Capella error message is hard to read) outweighs the chance that there will a new image format.